Creating data model

ZahraArshia opened the pull request:
Hello 😀 

## in this step of the project:

- [ ] project schema is built:

    - necessary migration files created
        - `rails generate model MODEL_NAME`
    - Table and column names created according to the ERD diagram(Foreign keys and their corresponding indexes included)

![image](https://user-images.githubusercontent.com/78906545/172895882-ff08d4a6-0242-4065-9f4b-c7ee6af45317.png)

## results :
![image](https://user-images.githubusercontent.com/78906545/172896990-39de4282-0518-4632-9df8-02cbca64504b.png)

ZahraArshia committed:
generate "user" model

ZahraArshia committed:
generate "post" model

ZahraArshia committed:
generate "comment" model

ZahraArshia committed:
generate "like" model

ZahraArshia committed:
run all migrations and generate "schema.rb"

ZahraArshia committed:
fix linter errors

SajjadAhmad14 commented on db/migrate/20220609154646_create_posts.rb:
```diff
@@ -0,0 +1,14 @@
+class CreatePosts < ActiveRecord::Migration[7.0]
+  def change
+    create_table :posts do |t|
+      t.belongs_to :users, null: false, foreign_key: true
```
Please create `author_id` as a foreign key instead of `user_id` as it can be seen here in [this](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png) ERD.

SajjadAhmad14 commented on db/migrate/20220609155556_create_comments.rb:
```diff
@@ -0,0 +1,12 @@
+class CreateComments < ActiveRecord::Migration[7.0]
+  def change
+    create_table :comments do |t|
+      t.belongs_to :users, null: false, foreign_key: true
```
Please create `author_id` as a foreign key instead of `user_id` as it can be seen here in [this](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png) ERD.

SajjadAhmad14 commented on db/migrate/20220609155843_create_likes.rb:
```diff
@@ -0,0 +1,10 @@
+class CreateLikes < ActiveRecord::Migration[7.0]
+  def change
+    create_table :likes do |t|
+      t.belongs_to :users, null: false, foreign_key: true
```
Please create `author_id` as a foreign key instead of `user_id` as it can be seen here in [this](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png) ERD.

SajjadAhmad14 reviewed (changes requested):
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi Zahra, 
 ### You have been doing a tremendous job so far :ok_hand:

## To Highlight 🎉
Git flow is followed. :heavy_check_mark:
README is professional. :heavy_check_mark:
## To Improve 🙏
- [ ] Please do not give the same names to all three databases in `databse.yml`. because it will create only one database which is the first one. But we need to create all three databases. Kindly see
![database](https://user-images.githubusercontent.com/35504149/172902105-921eaca2-0c3d-46a2-bd9a-4343e3bb6484.png)
 the screenshot below of your code.


- [ ] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
**_Note: Please do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

ZahraArshia committed:
replace author_Id with users_Id as foreign key

ZahraArshia committed:
create all three databases

ZahraArshia commented:
Hello,
thanks for the review 🌹 
- no `user-id` was used in the previse code, there was a “belogs_to” relationship between tables which was created automatically while creating models and since the authors were users, implicitly it was actually `AuthorId` itself !
so there was no `user-id` in the first place, but since the reviewer asked me to do so, I changed the structure of tables and implemented the references manually that required generating a separated migrate file:

![image](https://user-images.githubusercontent.com/78906545/172918807-0e0858bc-df5c-4e69-a28f-7677d03516d9.png)

- the `databse.yml` file modified as requested to include all three databases

Lordkaito reviewed (approved):
Hi @ZahraArshia, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
